ehrql/query_engines/mssql.py

```diff
@@ -173,9 +173,10 @@ def get_results(self, variable_definitions):
         setup_queries, cleanup_queries = get_setup_and_cleanup_queries(results_query)
 
         with self.engine.connect() as connection:
-            # Previously, when using the temporary database to store results, it was
-            # important to use AUTOCOMMIT. Possibly we don't need this anymore, but in
-            # the spirit of not changing too much at once I'm leaving it in place.
+            # All our queries are either (a) read-only queries against static data, or
+            # (b) queries which modify session-scoped temporary tables. For this reason
+            # we can avoid the overhead of transaction integrity and can run in "YOLO"
+            # isolation mode.
```

Comment:
The nittest of nits: I wonder if we should be clearer on language here.

"isolation mode" to me means the DB level isolation, e.g. READ COMMITTED, etc.  AIUI this is the client side DB-API transaction mode, which feels like it maybe needs a different name? Or does it also set a DB level isolation mode?

Reply:
It's definitely worth being as clear as we can on language. I don't fully understand all the details here, but I'll at least make sure I mirror SQLAlchemy's description exactly.